### DISCUSSION TITLE
ISO: Million dollar programmers

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [megabyte-labs/install.doctor#40] [Announcements]
Hey community, I am currently looking for big cheese developers to contribute to my little desktop setup script. I can offer nothing for your contributions except the glory of Valhalla (and the opportunity to see some really great apps and utilities should you decide to take ID for a spin).

**Comment by silopolis** [comment]
I'm more like an old :fr: stinky one :sweat_smile:  but if I can get enough mentoring to be useful, I'd be happy to give a hand...

**Reply by ProfessorManhattan** [reply]
Hey sure, I think want I want to do is have all of these CLI tools automatically install on the fly. So basically, I'm looking to write a piece of code that executes using a command not found hook that installs the program before running the command. If you could use the bash command not found hook and write up code to use `apt-get` and `dnf` (or anything that requires sudo) to install the CLI tool before using it, that'd be great.

The part I'm unsure of at this point is how to configure the app so that `apt-get` can run sudoless without granting total sudo privileges to a user.

Another thing you can do to help is to provide a way of running the script with the default settings on all the various OSes that I can hook into GitHub CI.
